.github/workflows/e2e-tests.yml

```diff
@@ -12,20 +12,87 @@ on:
         type: string
 
 jobs:
+
   e2e:
-    name: "E2E Tests"
+    name: E2E Tests - Shard ${{ matrix.shard }} of ${{ matrix.total_shards }}
+    runs-on: ubuntu-latest
+
+    strategy:
+        matrix:
+          shard: [1, 2, 3]
+          total_shards: [3]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+
+      - name: Run e2e tests (Shard ${{ matrix.shard }}/${{ matrix.total_shards }})
+        run: |
+          make e2e-test \
+            APP_NAME=${{ inputs.app_name }} \
+            BASE_URL=${{ inputs.service_endpoint }} \
+            CURRENT_SHARD=${{ matrix.shard }} \
+            TOTAL_SHARDS=${{ matrix.total_shards }}
+
+      - name: Verify blob report directory after tests
+        run: |
+          echo "Contents of blob-report directory on host:"
+          ls -R ./e2e/blob-report || echo "blob-report directory not found"
+
+      - name: Upload Blob Report
+        if: ${{ !cancelled() }}
```

Comment:
An extra safeguard that on manual or timeout-based cancellations, steps within `e2e` or `create-report` don't run - but I think since `create-report` has a dependency on `e2e` we don't need ths line - let's remove.

Reply:
They were originally using it in there merge and upload blob reports on the playwright sharding docs https://playwright.dev/docs/test-sharding#github-actions-example

Reply:
if you look at their example it's because they set fail-fast to false, but they still want to upload the blob even if the e2e tests failed, and they still want to merge even if some of the shards failed. seems like we'd want to do the same, and add a comment saying that "still upload report even if previous step failed"